tool/src/java/org/sakaiproject/gradebookng/business/GradebookNgBusinessService.java

```diff
@@ -552,8 +552,169 @@ public void updateAssignmentOrder(String siteId, long assignmentId, int order) {
 		Gradebook gradebook = this.getGradebook(siteId);
 		this.gradebookService.updateAssignmentOrder(gradebook.getUid(), assignmentId, order);
     }
-    
-    /**
+
+
+  /**
+   * Update the categorized order of an assignment.
+   *
+   * @param siteId	the siteId
+   * @param assignmentId the assignment we are reordering
+   * @param order the new order
+   * @throws JAXBException
+   * @throws IdUnusedException
+   * @throws PermissionException
+   */
+  public void updateCategorizedAssignmentOrder(String siteId, long assignmentId, int order) throws JAXBException, IdUnusedException, PermissionException {
+    Gradebook gradebook = (Gradebook)gradebookService.getGradebook(siteId);
+    Assignment assignmentToMove = gradebookService.getAssignment(gradebook.getUid(), assignmentId);
+    String category = assignmentToMove.getCategoryName();
```

Comment:
This may throw a NPE if null, null check is afterwards, probably needs to be before.